services/bettersocial/chat/allAnonChat.js

```diff
@@ -3,11 +3,12 @@ const {Op} = require('sequelize');
 const crypto = require('crypto');
 const BetterSocialConstantListUtils = require('../constantList/utils');
 
-const generate_channel_id_for_anon_chat = (owner, member, context = null) => {
+const generate_channel_id_for_anon_chat = (owner, member, context = null, source_id = null) => {
   const hash = crypto.createHash('sha256');
   hash.update(owner);
   hash.update(member);
   if (context) hash.update(context);
+  if (source_id) hash.update(source_id);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made to the `generate_channel_id_for_anon_chat` function seem fine. The addition of the `source_id` parameter and its subsequent use in the hash calculation process is consistent with the PR's description. However, it's important to ensure that the `source_id` is always a string before updating the hash with it. If it's not a string, it could potentially cause an error.

```diff
- if (source_id) hash.update(source_id);
+ if (source_id) hash.update(String(source_id));
```
This change ensures that even if a non-string value is passed as `source_id`, it will be converted to a string before being used in the hash update operation, preventing potential errors.


<!-- This is an auto-generated comment by OSS CodeRabbit -->